src/main/java/org/chiclepad/backend/Dao/CategoryDao.java

```diff
@@ -10,129 +9,73 @@
-//      String salt = resultSet.getString("salt");
-//      String name = resultSet.getString("name");
-//      String localeCode = resultSet.getString("locale");
-//      Locale locale = null;
-//
-//
-//
-//      return new Category(userId, email, password, salt);
-//   }
-//
-//   public List<Category> getAll() {
-//      String sqlGetAll = "SELECT * FROM category"
-//            + " LEFT OUTER JOIN category_details ON category_details.user_id = category.id;";
-//
-//      return jdbcTemplate.query(sqlGetAll,
-//            (RowMapper<Category>) (resultSet, rowNum) -> {
-//               return getCategory(resultSet);
-//            }
-//      );
-//   }
-//
-//   //UPDATE
-//   public Category update(Category category) throws DuplicateKeyException {
-////
-////      String sqlUpdatePassword = "UPDATE category "
-////            + "SET password = ? WHERE id = "
-////            + category.getId() + ";";
-////      jdbcTemplate.update(sqlUpdatePassword, category.getPassword());
-////
-////      String sqlAreUserDetailsPresent =
-////            " SELECT count(*) FROM category_details WHERE  user_id = " + category.getId() + " ;";
-////
-////      int userId = -1;
-////      if (category.getLocale().isPresent() || category.getName().isPresent()) {
-////         Integer userIdQueried = jdbcTemplate.queryForObject(sqlAreUserDetailsPresent, Integer.class);
-////         if (userIdQueried != null) {
-////            userId = userIdQueried;
-////         }
-////
-////         if (userId == 0) {
-////            String sqlInsertUserDetailsRow =
-////                  "INSERT INTO category_details(user_id) VALUES (" + category.getId() + ")";
-////            jdbcTemplate.update(sqlInsertUserDetailsRow);
-////         }
-////
-////         // Set locale
-////         if (category.getLocale().isPresent()) {
-////            Locale locale = category.getLocale().get();
-////            String sqlUpdateLocale =
-////                  "UPDATE category_details SET locale = ? WHERE user_id = "
-////                        + category.getId()
-////                        + ";";
-////            jdbcTemplate.update(sqlUpdateLocale, locale);
-////         }
-////
-////         // Set name
-////         if (category.getName().isPresent()) {
-////            String name = category.getName().get();
-////            String sqlUpdateName =
-////                  "UPDATE category_details SET name = ? WHERE user_id = "
-////                        + category.getId()
-////                        + ";";
-////            jdbcTemplate.update(sqlUpdateName, name);
-////         }
-////      }
-//
-//      return category;
-//   }
-//
-//   //DELETE
-//   public Category delete(Category category) {
-//      String sqlDelete = "DELETE  FROM category WHERE  id = " + category.getId();
-//      jdbcTemplate.update(sqlDelete);
-//      return category;
-//   }
+   // Data source
+   private JdbcTemplate jdbcTemplate;
 
+   public CategoryDao(JdbcTemplate jdbcTemplate) {
```

Comment:
Dao constructors can be `protected` to allow getting instances only using `DaoFactory`.